pkg/comp-functions/functions/vshnpostgres/user_management.go

```diff
@@ -38,17 +38,28 @@ func UserManagement(ctx context.Context, comp *vshnv1.VSHNPostgreSQL, svc *runti
 
 	for _, access := range comp.Spec.Parameters.Service.Access {
 
-		userPasswordRef := addUser(comp, svc, *access.User)
-
 		dbuser := *access.User
 		dbname := dbuser
 		if access.Database != nil {
 			dbname = *access.Database
 		}
 
-		addDatabase(comp, svc, dbuser, dbname)
+		userPasswordRef := addUser(comp, svc, *access.User)
+		roleResourceName := fmt.Sprintf("%s-%s-role", comp.GetName(), dbuser)
+
+		dbResourceName := fmt.Sprintf("%s-%s-database", comp.GetName(), dbname)
+		if svc.IsResourceSyncedAndReady(roleResourceName) {
+			addDatabase(comp, svc, dbuser, dbname)
+		} else {
+			return runtime.NewWarningResult(fmt.Sprintf("Role %s not ready, waiting before creating database %s", roleResourceName, dbResourceName))
```

Comment:
It's better to just log here and also add an event to the the state. Instead of returning, as it could drop stuff from the desired state.